controllers/tf_controller_runner.go

```diff
@@ -233,6 +234,8 @@ func (r *TerraformReconciler) runnerPodSpec(terraform infrav1.Terraform, tlsSecr
 	}
 
 	return v1.PodSpec{
+		Hostname:                      terraform.Name,
+		Subdomain:                     "tf-runner",
```

Comment:
Maybe we introduce `--use-pod-subdomain-resolution` to guard here.

Reply:
Yeah, will do !

Reply:
Those two fields are now guarded by `--use-pod-subdomain-resolution`